### PR TITLE
Edit admin users ld

### PR DESCRIPTION
**Pull request by LoganDemaray316**
Allows an admin to edit users with their roles and edits their information if they are students.

Edited Files:
* UserStore.js
* EditUsers.vue

Added Files:
* AdminUserComponent.vue
* EditSingleUser.vue

### CODE REVIEW

File: src/components/Admin/AdminUserComponent.vue [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+              <v-btn
+                @click="this.editSingleUser = true"
+                elevation="0"
+                size="small"
+                rounded="pill"
+                class="buttonWhite text-mediumBlue font-weight-bold">
+                Edit
+              </v-btn>
+              <v-dialog v-model="editSingleUser" max-width="600px">
+                <EditSingleUser
```
Comment (ethanimooney) [inline src/components/Admin/AdminUserComponent.vue:35]: I might rename these to "...Dialog"

File: src/stores/UserStore.js [hunk 2]
```diff
@@ -200,33 +194,46 @@ export const useUserStore = defineStore("user", {
...
-    // Post update to the DB, then update store
-    // This needs to be used carefully, it's not really safe to use yet
-    // @ethanimooney: Fix?
-    async updateUserInfo(data, role) {
-      if (role === "student") {
-        await this.updateStudentInfo(data);
-      } else if (role === "faculty") {
-        await updateFacultyInfo(data);
-      } else if (role === "admin") {
+    async updateUserInfo(data, role, id) {
```
Comment (ethanimooney) [inline src/stores/UserStore.js:197]: Is this being used?

Comment (ethanimooney) [reply src/stores/UserStore.js:197]: If not just remove it

If yes, it needs to be updated to check for roles correctly based on the new number system.

**Review by ethanimooney — CHANGES_REQUESTED** [review]

File: src/stores/UserStore.js (hunk 2 above)
Comment (LoganDemaray316) [reply src/stores/UserStore.js:197]: Fixed

File: src/components/Admin/AdminUserComponent.vue (hunk 1 above)
Comment (LoganDemaray316) [reply src/components/Admin/AdminUserComponent.vue:35]: Fixed

**Review by ethanimooney — APPROVED** [review]
Lgtm

**Pull request merged**